src/bcccp/carpark/exit/ExitController.java

```diff
@@ -6,149 +6,320 @@
 import bcccp.carpark.ICarpark;
 import bcccp.carpark.IGate;
 import bcccp.tickets.adhoc.IAdhocTicket;
```

Comment:
Need some imports:
import bcccp.carpark.ICarSensor;
+import bcccp.carpark.ICarSensorResponder;
+import bcccp.carpark.ICarpark;
+import bcccp.carpark.IGate;
+import bcccp.tickets.adhoc.IAdhocTicket;

Reply:
maybe you can simplify as 
```
import bcccp.carpark.*;
import bcccp.tickets.*;
```